src/api/v1alpha1/edgeconnect/edgeconnect_types.go

```diff
@@ -0,0 +1,157 @@
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EdgeConnectSpec defines the desired state of EdgeConnect
+type EdgeConnectSpec struct {
+	// Add custom validation using kubebuilder tags: https://book-v1.book.kubebuilder.io/beyond_basics/generating_crd.html
+
+	// Location of the Dynatrace API to connect to, including your specific environment UUID
+	// +required
+	// +kubebuilder:validation:Required
+	ApiServer string `json:"apiServer"`
+
+	// EdgeConnect uses the OAuth client to authenticate itself with the Dynatrace platform.
+	// +required
+	// +kubebuilder:validation:Required
+	OAuth OAuthSpec `json:"oauth"`
+
+	// Optional: restrict outgoing HTTP requests to your internal resources to specified hosts
+	// +kubebuilder:validation:Optional
+	HostRestrictions string `json:"hostRestrictions,omitempty"`
+
+	// Optional: image reference
+	// +kubebuilder:validation:Optional
+	ImageRef ImageRefSpec `json:"imageRef,omitempty"`
+
+	// Optional: enables automatic restarts of EdgeConnect pods in case a new version is available (the default value is: true)
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=true
+	AutoUpdate bool `json:"autoUpdate"`
+
+	// Optional: pull secret for your private registry
+	// +kubebuilder:validation:Optional
+	CustomPullSecret string `json:"customPullSecret,omitempty"`
+
+	// Optional: adds additional annotations for the EdgeConnect pods
+	// +kubebuilder:validation:Optional
+	Annotations map[string]string `json:"annotations,omitempty"`
+
+	// Optional: adds additional labels for the EdgeConnect pods
+	// +kubebuilder:validation:Optional
+	Labels map[string]string `json:"labels,omitempty"`
+
+	// Optional: adds additional environment variables for the EdgeConnect pods
+	// +kubebuilder:validation:Optional
+	Env []corev1.EnvVar `json:"env,omitempty"`
+
+	// Amount of replicas for your EdgeConnect (the default value is: 1)
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default:=1
+	Replicas *int32 `json:"replicas"`
+
+	// Optional: define resources requests and limits for single pods
+	// +kubebuilder:validation:Optional
+	Resources corev1.ResourceRequirements `json:"resources,omitempty"`
+
+	// Optional: node selector to control the selection of nodes for the EdgeConnect pods
+	// +kubebuilder:validation:Optional
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+
+	// Optional: set tolerations for the EdgeConnect pods
+	// +kubebuilder:validation:Optional
+	Tolerations []corev1.Toleration `json:"tolerations,omitempty"`
+
+	// Optional: set topology spread constraints for the EdgeConnect pods
+	// +kubebuilder:validation:Optional
+	TopologySpreadConstraints []corev1.TopologySpreadConstraint `json:"topologySpreadConstraints,omitempty"`
+}
+
+type OAuthSpec struct {
+	// The secret from your OAuth client generation
+	// +required
+	// +kubebuilder:validation:Required
+	ClientSecret string `json:"clientSecret"`
```

Comment:
slightly different wording in the [edgeconnect doc](https://docs.dynatrace.com/platform/shortlink/edgeconnect#config), please align that.

We also don't need the `The` at the start

Reply:
fixed